components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/ServiceURLBuilderFactory.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.core;
+
+import org.wso2.carbon.identity.core.internal.DefaultServiceURLBuilder;
+
+/**
+ * Service URL Builder Factory representation.
+ */
+public class ServiceURLBuilderFactory {
+
+    /**
+     * This method is called to create a {@link ServiceURLBuilder} instance.
+     *
+     * @return {@link ServiceURLBuilder}.
+     */
+    public ServiceURLBuilder getServiceURLBuilder() {
```

Comment:
As this a factory wouldn't it be better if we say createServiceURLBuilder, else it does not denote at once that a new builder will be created

Reply:
open for discussions @mefarazath